From 8b0fb6f872477dfd218e43b073543e4913b6d785 Mon Sep 17 00:00:00 2001
From: Dan Carpenter <error27@gmail.com>
Date: Thu, 10 Mar 2011 11:31:33 +0300
Subject: USB: ene_ub6250: fix memory leak in ene_load_bincode()
Patch-mainline: v2.6.39
Git-commit: 8b0fb6f872477dfd218e43b073543e4913b6d785

"buf" gets allocated twice in a row.  It's the second allocation which
is correct.  The first one should be removed.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Acked-by: huajun li <huajun.li.lee@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/ene_ub6250.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/usb/storage/ene_ub6250.c
+++ b/drivers/usb/storage/ene_ub6250.c
@@ -491,10 +491,6 @@ static int ene_load_bincode(struct us_da
 	if (info->BIN_FLAG == flag)
 		return USB_STOR_TRANSPORT_GOOD;
 
-	buf = kmalloc(ENE_BIN_CODE_LEN, GFP_KERNEL);
-	if (buf == NULL)
-		return USB_STOR_TRANSPORT_ERROR;
-
 	switch (flag) {
 	/* For SD */
 	case SD_INIT1_PATTERN:
